enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,587 @@
+- As a customer I want documentation about how to perform the backup and recovery procedures in my Hosted Cluster.
+
+### Goals
+
+- Provide a consumable API for Etcd backup that could be usable in a Hosted Cluster restoration scenario. These are the features included:
+  - Schedulable.
+  - Capability to select the right Time Zone.
+  - Retention policy based on size of the backup or number of backups.
+  - Local storage of the backup (PVCs).
+  - Remote storage of the backup in a service provider (S3, Azure...).
+  - Restoration of the backup in another Hosted Cluster using the already existent API.
+
+### Non-Goals
+
+- Provide a full backup/restoration solution for a hosted cluster.
+  - This is not covering the Volumes migrations
+
+
+### Pros and Cons
+#### Pros
+
+**Visibility:** Having an API could enhance visibility within the HostedCluster (HC) object, which serves as the primary interface for customers to manage backups. In the event of a backup failure, in addition to log entries, it could update the status.conditions within the HostedCluster object.
+
+**Integration:** Integration with the product is strengthened, meaning that adding new features becomes more straightforward. Employing tightly coupled methods could lead to poor practices.
+
+**Alerting:** The controller can notify users through various channels:
+
+- HC status.conditions
+- Log entries
+- Events
+- Alerting
+- Metrics
+
+**Validations:** In this scenario, we incorporate an additional layer of validations (CEL), managed by api-machinery. This layer promptly detects any discrepancies with the API consensus and triggers early failure. Furthermore, additional validations can be employed within the controller.
+
+**Reusability:** Leveraging the existing ETCD backup API developed by the ETCD team, we will integrate the entire API into a new structure within our ETCD API group. Within the same group, we will incorporate the necessary functionality to upload binaries to remote storage such as S3 or Azure.
+
+#### Cons
+
+**Supportability:** Development of the API and support of it during the product life-cycle.
+**Complexity:** The Hosted Control Planes team will take the responsibility to stay tuned over the API changes of the ETCD Backup.
+
+
+## Proposal
+
+The proposal commences with the acknowledgement that a user requires an ETCD backup.
+
+The primary requirement is that the ETCD `managementType` must be configured as `managed`. Once this configuration is set, further progress can proceed.
+
+This backup operation will be specified within the `hostedcluster.spec.etcd` API group under a new subgroup named `backup`. This subgroup will encompass two fields, with the primary one designated for the backup process itself, directly imported from the original ETCD Backup API. Comprehensive details regarding this API can be found in the official ETCD proposal and its corresponding implementation documentation.
```

Comment:
I don't think the terms API group and subgroup is correctly used here. we only have one API group `hypershift.openshift.io`.

`hostedcluster.spec.etcd` is just a field within the HostedCluster kind.